packages/ui/src/components/organisms/chart/chart.utils.ts

```diff
@@ -0,0 +1,188 @@
+import { ChartInterval, ChartTickInterval } from "./chart.types";
+
+type DateParams = number | string | Date;
+
+/**
+ * Downsamples data to a maximum number of data points.
+ * Useful for displaying large datasets in charts to avoid performance issues.
+ */
+export const getDownsampledData = <T>(
```

Comment:
I think this could potentially lose the last data point if length isn't divisible by interval, and it wouldn't show the peaks and valleys very well. How hard would it be to implement the Largest Triangle Three Bucket algorithm?

https://github.com/seanvelasco/lttb/blob/main/src/index.ts